Apps/Common/src/Services/AuditService.cs

```diff
@@ -39,5 +53,73 @@ public Task WriteAuditEvent(AuditEvent auditEvent)
             
             return t;
         }
+
+        /// <inheritdoc />
+        public AuditEvent ParseHttpContext(HttpContext context, AuditEvent audit)
+        {
+            audit.ApplicationName = this.getApplication();
+            audit.TransactionResult = this.getTransactionResult(context.Response.StatusCode);
+            audit.ApplicationSubject = context.User.Identity.Name;
+            audit.CreatedBy = string.IsNullOrEmpty(context.User.Identity.Name) ?
+                ANONYMOUS : context.User.Identity.Name;
+            audit.TransacationName = context.Request.Path;
+            audit.Trace = context.TraceIdentifier;
+            audit.ClientIP = context.Connection.RemoteIpAddress.MapToIPv4().ToString();
+
+            RouteData routeData = context.GetRouteData();
+            audit.TransactionVersion = routeData != null ? routeData.Values["version"].ToString() : "";
+            
+            return audit;
+        }
+
+        /// <summary>
+        /// Gets the transaction result from the http context status code.
+        /// </summary>
+        /// <param name="statusCode">The http context status code.</param>
+        /// <returns>The mapped transaction result.</returns>
+        private AuditTransactionResult getTransactionResult(int statusCode)
+        {
+            // Success codes (1xx 2xx 3xx)
+            if (statusCode < 400)
+            {
+                return AuditTransactionResult.Success;
+            }
+
+            // Unauthorized and forbidden
+            if (statusCode == 401 || statusCode == 403)
+            {
+                return AuditTransactionResult.Unauthorized;
+            }
+
+            // Client/Request errors other than unauthorized and forbidden
+            if (statusCode >= 400 && statusCode < 500)
+            {   
+                return AuditTransactionResult.Failure;
+            } 
+            
+            return AuditTransactionResult.SystemError;
+        }
+
+        /// <summary>
+        /// Gets the current application.
+        /// </summary>
+        /// <returns>The mapped application.</returns>
+        private Applications getApplication()
+        {
+            AssemblyName assemblyName = Assembly.GetEntryAssembly().GetName();
+            switch (assemblyName.Name)
+            {
+                case "Immunization":
```

Comment:
Let's create constants for this. A new file with constants would also host ones for different projects

Reply:
Refactored the code to parse it and map it directly to the enum instead of constants or hd strings